pkg/internal/tests/cross-tests/tfwrite.go

```diff
@@ -16,115 +16,70 @@
 package crosstests
 
 import (
-	"fmt"
-	"io"
-	"sort"
-
-	"github.com/hashicorp/hcl/v2/hclwrite"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
-	"github.com/zclconf/go-cty/cty"
+
+	crosstestsimpl "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/internal/tests/cross-tests/impl"
 )
 
-// Provider writes a provider declaration to an HCL file.
-//
-// Note that unknowns are not yet supported in cty.Value, it will error out if found.
-func writeProvider(out io.Writer, sch map[string]*schema.Schema, typ string, config cty.Value) error {
-	if !config.IsWhollyKnown() {
-		return fmt.Errorf("WriteHCL cannot yet write unknowns")
+func sdkV2NestingToShim(nesting string) crosstestsimpl.Nesting {
```

Comment:
This should take and switch on a schema.Nesting, not it's string representation.
```suggestion
func sdkV2NestingToShim(nesting schema.NestingMode) crosstestsimpl.Nesting {
```

Reply:
The schema.NestingMode is a private type, so we can't use it as a parameter here

Reply:
I'll use the same method I used in PF where we copied the type over.